library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -2638,8 +2687,8 @@ fn count(tail: usize, head: usize, size: usize) -> usize {
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
-impl<A: PartialEq> PartialEq for VecDeque<A> {
-    fn eq(&self, other: &VecDeque<A>) -> bool {
+impl<T: PartialEq, A: Allocator> PartialEq for VecDeque<T, A> {
+    fn eq(&self, other: &Self) -> bool {
```

Comment:
Ditto: why require the same allocator? Is it required by the trait definition?

Reply:
This is not required by the definition (only for `Eq`), but [this is what `Vec` does](https://doc.rust-lang.org/nightly/alloc/vec/struct.Vec.html#impl-PartialEq%3CVec%3CU%2C%20A%3E%3E), so I did the same.